src/Algebra/Graph/NonEmpty.hs

```diff
@@ -212,8 +212,8 @@ ordInt :: Graph Int -> Graph Int -> Ordering
 ordInt x y = compare (T.toAdjacencyIntMap x) (T.toAdjacencyIntMap y)
 
 instance Applicative Graph where
-    pure  = Vertex
-    (<*>) = ap
+    pure    = Vertex
+    f <*> x = f >>= (<$> x)
```

Comment:
```suggestion
    f <*> x = apR f x
```

Why not this? Seems simpler and less code duplication. But perhaps this makes rewriting more difficult?

Reply:
I just not implemented all the rewrite rules system for `Graph.NonEmpty`, so `<*>` will anyway not be rewritten :p

Reply:
Oh, sorry, my bad :)